zokrates_core/src/embed.rs

```diff
@@ -281,6 +321,102 @@ pub fn sha256_round<T: Field>() -> FlatFunction<T> {
     }
 }
 
+#[cfg(feature = "ark")]
+pub fn verify<T: Field>(n: usize) -> FlatFunction<T> {
+    let (out_index, input_indices, proof_indices, vk_indices, constraints, variable_count) =
+        generate_verify_constraints(n);
+
+    let cs_indices = 0..variable_count;
+    let input_indices = input_indices.into_iter();
+    let proof_indices = proof_indices.into_iter();
+    let vk_indices = vk_indices.into_iter();
+
+    // indices of the arguments to the function
+    let input_argument_indices = input_indices.clone().map(|i| i + variable_count);
+
+    let proof_argument_indices = proof_indices.clone().map(|i| i + variable_count);
+
+    let vk_argument_indices = vk_indices.clone().map(|i| i + variable_count);
+
+    let input_arguments = input_argument_indices
+        .clone()
+        .map(|i| FlatParameter::private(FlatVariable::new(i)));
+
+    let proof_arguments = proof_argument_indices
+        .clone()
+        .map(|i| FlatParameter::private(FlatVariable::new(i)));
+
+    let vk_arguments = vk_argument_indices
+        .clone()
+        .map(|i| FlatParameter::private(FlatVariable::new(i)));
+
+    let arguments = input_arguments
+        .chain(proof_arguments)
+        .chain(vk_arguments)
+        .collect();
+
+    let one_binding_statement = FlatStatement::Condition(
+        FlatExpression::Identifier(FlatVariable::new(0)),
```

Comment:
Did you check that ark uses variable #0 as ~one?

Reply:
It was the case for sha256 in bellman but worth double checking

Reply:
should be `0` judging by this
https://github.com/arkworks-rs/snark/blob/e9496ffa7852cca5775e0fdaa6062677d1444381/relations/src/r1cs/mod.rs#L114